src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -276,4 +280,48 @@ public ResponseEntity<Result> createResultForExternalSubmission(@PathVariable Lo
         return ResponseEntity.created(new URI("/api/results/" + savedResult.getId()))
                 .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, savedResult.getId().toString())).body(savedResult);
     }
+
+    /**
+     * GET exercises/:exerciseId/feedback-details : Retrieves all negative feedback details and the latest result IDs for a given exercise.
+     *
+     * @param exerciseId The ID of the exercise for which feedback details and result IDs should be retrieved.
+     * @return A ResponseEntity containing a list of all feedback details and the corresponding result IDs for the exercise.
+     */
+    @GetMapping("exercises/{exerciseId}/feedback-details")
+    @EnforceAtLeastTutorInExercise
+    public ResponseEntity<FeedbackDetailsWithResultIdsDTO> getAllFeedbackDetailsForExercise(@PathVariable Long exerciseId) {
+        log.debug("REST request to get all Feedback details for Exercise {}", exerciseId);
+
+        Set<StudentParticipation> participations = studentParticipationRepository.findByExerciseIdWithLatestResultsAndFeedbackAndTestcasesElseThrow(exerciseId);
+        removeSubmissionAndExerciseData(participations);
+
+        List<FeedbackDetailDTO> allFeedbackDetails = new ArrayList<>();
+        List<Long> resultIds = new ArrayList<>();
+
+        participations.forEach(participation -> {
+            participation.getResults().forEach(result -> {
+                resultIds.add(result.getId());
+                result.getFeedbacks().forEach(feedback -> {
+                    if (Boolean.FALSE.equals(feedback.isPositive())) {
+                        String detailText = feedback.getDetailText();
+                        String testCaseName = feedback.getTestCase() != null ? feedback.getTestCase().getTestName() : null;
+                        FeedbackDetailDTO feedbackDetailDTO = new FeedbackDetailDTO(detailText, testCaseName);
+                        allFeedbackDetails.add(feedbackDetailDTO);
+                    }
+                });
+            });
+        });
+
+        FeedbackDetailsWithResultIdsDTO response = new FeedbackDetailsWithResultIdsDTO(allFeedbackDetails, resultIds);
+        return new ResponseEntity<>(response, HttpStatus.OK);
```

Comment:
Optional: A little simpler:
```suggestion
        return new ResponseEntity.ok(response);
```